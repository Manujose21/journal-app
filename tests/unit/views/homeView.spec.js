import HomeView from '@/views/HomeView.vue'
import { shallowMount } from '@vue/test-utils'

describe('tests home view ', ()=>{
  it('snapshot component', ()=>{
    const wrapper = shallowMount(HomeView)

    expect(wrapper.html()).toMatchSnapshot()

  })

  it('router click button', ()=>{
   
    const mockRouter = {
      push: jest.fn()
    }
    
    const wrapper = shallowMount(HomeView, {
      global:{
        mocks:{
          $router: mockRouter
        }
      }
    })
    
    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith( { name: "not-entry" } )
    
  })
})