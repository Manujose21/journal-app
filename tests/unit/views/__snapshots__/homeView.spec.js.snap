// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests home view  snapshot component 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <div><button class="btn btn-primary">primary</button><button class="btn btn-success">succes</button><button class="btn btn-secondary">secondary</button></div>
</div>
`;
